src/H5Ipkg.h

```diff
@@ -59,7 +59,10 @@ typedef struct H5I_id_info_t {
     hid_t       id;        /* ID for this info */
     unsigned    count;     /* Ref. count for this ID */
     unsigned    app_count; /* Ref. count of application visible IDs */
-    const void *object;    /* Pointer associated with the ID */
+    union {
```

Comment:
Allow an ID's object to be treated as 'const' or not 'const' as needed, avoiding warnings / diagnostic suppression noise in H5I code.